pkg/vsphere/adapter_test.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2020 VMware, Inc.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package vsphere
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"testing"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/client"
+	cecontext "github.com/cloudevents/sdk-go/v2/context"
+	"github.com/cloudevents/sdk-go/v2/event"
+	cehttp "github.com/cloudevents/sdk-go/v2/protocol/http"
+	"github.com/google/go-cmp/cmp"
+	"github.com/vmware/govmomi/vim25/types"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest"
+)
+
+type roundTripperTest struct {
+	statusCodes  []int
+	requestCount int
+	events       []*event.Event
+}
+
+func (r *roundTripperTest) RoundTrip(req *http.Request) (*http.Response, error) {
+	code := r.statusCodes[r.requestCount]
+	msg := cehttp.NewMessageFromHttpRequest(req)
+	e, err := binding.ToEvent(context.TODO(), msg)
+	if err != nil {
+		return nil, err
+	}
+	r.events = append(r.events, e)
+	return &http.Response{StatusCode: code}, nil
+}
+
+type mockType struct {
```

Comment:
Do we need that dedicated mock type (instead of using the `BaseEvent` interface) or is this just following Knative test conventions?

Reply:
Well, I wasn't quite sure what all we wanted to mock. It implements the BaseEvent interface so figured we can control these little more? Happy to change it if gives you heebyjeebies :)

Reply:
Yeah, I wasn't sure whether there's more to come than purely wrapping a `BaseEvent` type. Makes sense.